From: Shengjing Zhu <zhsj@debian.org>
Date: Sat, 6 Jan 2024 21:25:30 +0800
Subject: Skip TestUser{Cache,Config}Dir when HOME is not writable

---
 src/os/os_test.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/os/os_test.go b/src/os/os_test.go
index 2f5b117..2a3ada8 100644
--- a/src/os/os_test.go
+++ b/src/os/os_test.go
@@ -2848,6 +2848,10 @@ func TestUserCacheDir(t *testing.T) {
 	}
 
 	if err := MkdirAll(dir, 0777); err != nil {
+		if IsPermission(err) {
+			t.Log(err)
+			return
+		}
 		t.Fatalf("could not create UserCacheDir: %v", err)
 	}
 	d, err := MkdirTemp(dir, "TestUserCacheDir")
@@ -2871,6 +2875,10 @@ func TestUserConfigDir(t *testing.T) {
 	}
 
 	if err := MkdirAll(dir, 0777); err != nil {
+		if IsPermission(err) {
+			t.Log(err)
+			return
+		}
 		t.Fatalf("could not create UserConfigDir: %v", err)
 	}
 
