--- a/src/os/signal/signal_cgo_test.go
+++ b/src/os/signal/signal_cgo_test.go
@@ -27,7 +27,21 @@
 	"time"
 )
 
+// Check if we are in a chroot by checking if the inode of / is
+// different from 2 (there is no better test available to non-root on
+// linux).
+func isChrooted(t *testing.T) bool {
+	root, err := os.Stat("/")
+	if err != nil {
+		t.Fatalf("cannot stat /: %v", err)
+	}
+	return root.Sys().(*syscall.Stat_t).Ino != 2
+}
+
 func TestTerminalSignal(t *testing.T) {
+	if isChrooted(t) {
+		t.Skip("test cannot run in a chroot")
+	}
 	const enteringRead = "test program entering read"
 	if os.Getenv("GO_TEST_TERMINAL_SIGNALS") != "" {
 		var b [1]byte
