From: Shengjing Zhu <zhsj@debian.org>
Date: Sun, 11 Jun 2023 17:10:49 +0800
Subject: skip asan test on riscv64

Bug in GCC, see https://github.com/golang/go/issues/57691
---
 misc/cgo/testsanitizers/asan_test.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/misc/cgo/testsanitizers/asan_test.go b/misc/cgo/testsanitizers/asan_test.go
index 932cfb1..0233fde 100644
--- a/misc/cgo/testsanitizers/asan_test.go
+++ b/misc/cgo/testsanitizers/asan_test.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build !riscv64
+
 package sanitizers_test
 
 import (
