Index: golang-2011.04.13/src/cmd/8l/asm.c
===================================================================
--- golang-2011.04.13.orig/src/cmd/8l/asm.c	2011-04-28 09:41:46.000000000 +0200
+++ golang-2011.04.13/src/cmd/8l/asm.c	2011-04-28 09:45:44.000000000 +0200
@@ -40,7 +40,7 @@
 #define	Dbufslop	100
 
 char linuxdynld[] = "/lib/ld-linux.so.2";
-char freebsddynld[] = "/usr/libexec/ld-elf.so.1";
+char freebsddynld[] = "/lib/ld.so.1";
 
 int32
 entryvalue(void)
Index: golang-2011.04.13/src/cmd/6l/asm.c
===================================================================
--- golang-2011.04.13.orig/src/cmd/6l/asm.c	2011-04-28 09:42:26.000000000 +0200
+++ golang-2011.04.13/src/cmd/6l/asm.c	2011-04-28 09:45:25.000000000 +0200
@@ -42,7 +42,7 @@
 #define PADDR(a)	((uint32)(a) & ~0x80000000)
 
 char linuxdynld[] = "/lib64/ld-linux-x86-64.so.2";
-char freebsddynld[] = "/libexec/ld-elf.so.1";
+char freebsddynld[] = "/lib/ld-kfreebsd-x86-64.so.1";
 
 char	zeroes[32];
 
