--- a/src/cmd/8l/asm.c
+++ b/src/cmd/8l/asm.c
@@ -38,7 +38,7 @@
 #include	"../ld/pe.h"
 
 char linuxdynld[] = "/lib/ld-linux.so.2";
-char freebsddynld[] = "/usr/libexec/ld-elf.so.1";
+char freebsddynld[] = "/lib/ld.so.1";
 char openbsddynld[] = "/usr/libexec/ld.so";
 char netbsddynld[] = "/usr/libexec/ld.elf_so";
 
--- a/src/cmd/6l/asm.c
+++ b/src/cmd/6l/asm.c
@@ -40,7 +40,7 @@
 #define PADDR(a)	((uint32)(a) & ~0x80000000)
 
 char linuxdynld[] = "/lib64/ld-linux-x86-64.so.2";
-char freebsddynld[] = "/libexec/ld-elf.so.1";
+char freebsddynld[] = "/lib/ld-kfreebsd-x86-64.so.1";
 char openbsddynld[] = "/usr/libexec/ld.so";
 char netbsddynld[] = "/libexec/ld.elf_so";
 
